src/spring-cloud/azext_spring_cloud/_validators.py

```diff
@@ -73,6 +75,18 @@ def _validate_terms(cmd, namespace):
                                         '--plan tanzu-asc-ent-mtr" to accept the term.')
 
 
+def _check_tanzu_components_not_enable(cmd, namespace):
+    suffix = 'can only be used for Azure Spring Cloud Enterprise. Please add --sku="Enterprise" to create Enterprise instance.'
+    if namespace.enable_application_configuration_service:
+        raise InvalidArgumentValueError('--enable-application-configuration-service {}'.format(suffix))
+    if namespace.enable_service_registry:
+        raise InvalidArgumentValueError('--enable-service-registry {}'.format(suffix))
+    if namespace.enable_gateway:
+        raise InvalidArgumentValueError('--enable-gateway {}'.format(suffix))
+    if namespace.enable_api_portal:
+        raise InvalidArgumentValueError('--enable-api-portal {}'.format(suffix))
```

Comment:
`ArgumentUsageError`  may be the more appropriate error type, right?

Reply:
Updated.